### ISSUE TITLE
Common Accept headers takes precedence over the request's Accept headers.

### ISSUE THREAD

**Issue by JonWallsten** [axios/axios#1819]
#### Summary

I'm trying to override the Accept headers for a request. However, for some reason the Accept headers I add ends upp AFTER the Accept headers found in Axios.defaults.headers.common and hence I get the data back in the wrong format from our API.

I've tried to create a new instance of Axios and passed in the config with headers.common set to an empty object and null. I've tried to set headers.common.Accept to null and empty string. 
Whatever I do it breaks with either the default Accept headers still being there or that they are invalid because "null, " or ", " gets added to them.

The only way I found out so far that works is to either delete the Axios.defaults.headers.common or override it with whatever I want to set. I think it's weird that the headers I provide ends up with lower priority that the default ones. It makes no sense.

#### Context

- axios version: *0.18.0*
- Environment: *node v10.11.0, chrome 69, windows 10*

**Comment by danielpgauer** [comment]
+1 same problem

**Comment by buzzpsych** [comment]
+1

**Comment by tranceyos2419** [comment]
+1 having the same issue

**Comment by onemartini** [comment]
+1

**Comment by arielcamino** [comment]
+1

I think if you define an "accept" header it's because you **don't** want any random precedent header, no idea why this happens.

**Comment by lucsrods** [comment]
+1

**Comment by Alanscut** [comment]
That's weired, I tested with follow code, and it's ok. Could you provide more information?
```
os: win7&win10
version: v0.18.0
```
```
axios.defaults.headers.common["Accept"] = "aaaaa";  // low priority

var instance = axios.create({
  headers: {
    'Accept': 'fffff'  // middle priority
  }
})
instance.get('/', {
  headers: {
    'Accept': 'wwwww' // high  priority
  }
}).then(function ads(res) {
})
```

**Comment by chinesedfan** [comment]
Closed in favor of #2623.

**Comment by nicbou** [comment]
This can also happen if you set the `accept` header instead of the `Accept` header. In the first case, the value is appended to the default. In the second case, the value is replaced.
